psi4/src/psi4/libqt/CMakeLists.txt

```
@@ -24,6 +24,16 @@ list(APPEND sources
   timer.cc
   )
 
+# cmake_symbol_exists() or check_cxx_source_runs() are preferable to check_function_exists()
+#   according to CMake docs. But both require an explicit header, either in the call or in the code,
+#   and that's tough with multiple BLAS/LAPACK backends. So, we'll see if this is robust enough.
+include(CMakePushCheckState)
+include(CheckFunctionExists)
+cmake_push_check_state()
+list(APPEND CMAKE_REQUIRED_LIBRARIES tgt::blas)
+check_function_exists(daxpby _has_daxpby)
+cmake_pop_check_state()
+
```

Maybe adding a warning when DAXPBY is not detected would be a good idea? If detection break for whatever reason in the future, this would give people a chance to notice it eventually.
```suggestion
if(NOT _has_daxpby)
  message(WARNING "${Yellow}Your LAPACK library does not seem to be providing the DAXPBY subroutine. If you are seeing this message, and you are not building with Apple Accelerate, you may want to re-check that the correct BLAS/LAPACK library is being used by the build system.${ColourReset}")
endif()

```

---

good idea, will do.

> I actually would not mind at all if having a Fortran compiler would become an essential build-time dependency.

This probably isn't going to happen because it's unnecessarily burdensome (e.g., Mac gfortran was/is poorly supported for a while, and Win fortran compilers are half-a-dozen and none does it all and isn't proprietary, at least to my acquaintance). It's easier to get started if req'd deps are minimal (numpy, qcengine, optking, libint, libxc, gau2grid). Also, there's enough devs around who remember the novelty of a C-based QC program that they're not going to want required Fortran.

Nevertheless, I consider Fortran not a required dep but certainly a commonplace one so that addons can be in any language.

---

> > I actually would not mind at all if having a Fortran compiler would become an essential build-time dependency.
> 
> This probably isn't going to happen because it's unnecessarily burdensome (e.g., Mac gfortran was/is poorly supported for a while, and Win fortran compilers are half-a-dozen and none does it all and isn't proprietary, at least to my acquaintance). It's easier to get started if req'd deps are minimal (numpy, qcengine, optking, libint, libxc, gau2grid). Also, there's enough devs around who remember the novelty of a C-based QC program that they're not going to want required Fortran.

Yeah, I also prefer programs that keep the required deps few and easy, and I am not a huge fan of Fortran either. Still, in this case, Psi4 is paying the price in the C/Fortran interface complexity, and brittleness. The minimal-diff change that could cure that is probably to use CBLAS/LAPACKE, instead of directly calling the Fortran subroutines, and this would not require a Fortran compiler.
But I think we are getting a bit too off-topic for this PR.